arch/arc/core/reset.S

```diff
@@ -126,14 +126,17 @@ done_cache_invalidate:
         jl @_sys_resume_from_deep_sleep
 #endif
 
-#if CONFIG_MP_NUM_CPUS  > 1
+#if defined(CONFIG_SMP) || CONFIG_MP_NUM_CPUS  > 1
 	_get_cpu_id r0
```

Comment:
Does it make sense to have CONFIG_SMP defined with CONFIG_MP_NUM_CPUS = 1?
Wasn't that introduced only as a workaround for some other bug in the kernel mp testcase, and not really needed.

Reply:
in lots of examples and tests, there is CONFIG_SMP and CONFIG_MP_NUM_CPUS == 1.  Some times it makes sense that the target supports SMP but we just want to use one core. @andyross @andrewboie  what's your opinion?

Reply:
You would never do that for a product, no.  But SMP code paths are different in a few spots, so IMHO it's good from a code coverage perspective to have it enabled even if the test code itself isn't MP-safe.